src/effects.c

```diff
@@ -835,6 +837,25 @@ void libxmp_process_fx(struct context_data *ctx, struct channel_data *xc, int ch
 	case FX_FLT_RESN:
 		xc->filter.resonance = fxp;
 		break;
+	case FX_MACRO_SET:
+		xc->macro.active = LSN(fxp);
+		break;
+	case FX_MACRO:
+		SET(MIDI_MACRO);
+		xc->macro.val = fxp;
+		xc->macro.slide = 0;
+		break;
+	case FX_MACROSMOOTH:
+		if (ctx->p.speed && xc->macro.val < 0x80) {
+			SET(MIDI_MACRO);
+			xc->macro.target = fxp;
+			xc->macro.slide = round((fxp - xc->macro.val) / (double)ctx->p.speed);
```

Comment:
We should make `libxmp_round` in period.c public to the rest of the source and use it here.

Reply:
I'm not sure this `round` is actually required at all... this was one of the earlier parts I implemented a few weeks ago I and forgot why I did that. I can test further but I'm going to try to finish up a fuzz bugfix patch and the Digital Symphony patch first.

Reply:
Since values here are never negative you could just as well go with +0.5 and truncation.